CRM/Iats/Utils.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+class CRM_Iats_Utils {
+
+  public static function getSettings(): array {
+    $settingValues = [];
+    $settings = ['email_recurring_failure_report', 'bcc_email_recurring_failure_report', 'receipt_recurring', 'recurring_failure_threshhold', 'email_failure_contribution_receipt', 'disable_cryptogram', 'ach_category_text', 'no_edit_extra', 'enable_update_subscription_billing_info', 'enable_change_subscription_amount', 'enable_cancel_recurring', 'enable_cancel_recurring', 'days'];
+    $hasNotbeenMigrated = CRM_Core_DAO::singleValueQuery("SELECT id FROM civicrm_setting WHERE name = 'iats_settings'");
+    if ($hasNotbeenMigrated) {
+      return Civi::settings()->get('iats_settings');
+    }
+    else {
+      foreach ($settings as $setting) {
+        $settingValue = Civi::settings()->get('iats_' . $setting);
+        if ($setting === 'days') {
+          // if setting value is blank set it to be disabled.
+          if (trim($settingValue, CRM_Core_DAO::VALUE_SEPARATOR) === '') {
```

Comment:
JFYI, @seamuslee001, circa March 22, we started seeing test-failures like this with the git-based test-runs of `com.iatspayment.civicrm` (across all tested versions of `civicrm-core` and `php`):

```
CRM_Iats_SettingsTest::testGetDates
trim() expects parameter 1 to be string, array given

/home/homer/buildkit/build/build-0/web/sites/all/modules/civicrm/ext/com.iatspayments.civicrm/CRM/Iats/Utils.php:17
/home/homer/buildkit/build/build-0/web/sites/all/modules/civicrm/ext/com.iatspayments.civicrm/tests/phpunit/CRM/Iats/SettingsTest.php:56
/home/homer/buildkit/extern/phpunit9/phpunit9.phar:2307
```

[link](https://test.civicrm.org/job/CiviCRM-Ext-Matrix/BKPROF=min,BUILDTYPE=git,CIVIVER=master,EXTKEY=com.iatspayments.civicrm,label=bknix-tmp/4830/testReport/(root)/CRM_Iats_SettingsTest/testGetDates/)